betty/plugin/__init__.py

```diff
@@ -53,6 +54,20 @@ class Plugin(ABC):
     To test your own subclasses, use :py:class:`betty.test_utils.plugin.PluginTestBase`.
     """
 
+    @classmethod
+    @abstractmethod
+    def plugin_type_id(cls) -> MachineName:
```

Comment:
Introduce `PluginType`, add these methods there, and add just a single method to `Plugin` to get the type.